src/Gui/PythonWrapper.cpp

```diff
@@ -156,6 +157,38 @@ PyTypeObject** SbkPySide6_QtUiToolsTypes = nullptr;
 #include "UiLoader.h"
 #include "MetaTypes.h"
 
+#ifdef HAVE_SHIBOKEN
+#ifdef HAVE_SHIBOKEN2
+PyTypeObject** SbkPySide2_QtCoreTypes           = nullptr;
+PyTypeObject** SbkPySide2_QtGuiTypes            = nullptr;
+PyTypeObject** SbkPySide2_QtWidgetsTypes        = nullptr;
+PyTypeObject** SbkPySide2_QtPrintSupportTypes   = nullptr;
+PyTypeObject** SbkPySide2_QtUiToolsTypes        = nullptr;
+#define SbkPySide_QtCoreTypes           SbkPySide2_QtCoreTypes
+#define SbkPySide_QtGuiTypes            SbkPySide2_QtGuiTypes
+#define SbkPySide_QtWidgetsTypes        SbkPySide2_QtWidgetsTypes
+#define SbkPySide_QtPrintSupportTypes   SbkPySide2_QtPrintSupportTypes
+#define SbkPySide_QtUiToolsTypes        SbkPySide2_QtUiToolsTypes

```

Comment:
Can you explain this to me? Why do we need to use the preprocessor to do this?

Reply:
Later in the code, we are using the very same load function to load module. `SbkPySide2_*` and `SbkPySide6_*`  need to be globally visible symbols, otherwise build ends with undefined reference. Here we just define the same name despite fact we are using PySide2, PySide6 or Python access. In the last case dummy variable is used. All this to use as much common code as possible.

Reply:
Sorry my question was not clear: I meant, why specifically the preprocessor, rather than just using variables, e.g.
```

auto &SbkPySide_QtCoreTypes = SbkPySide2_QtCoreTypes;
auto &SbkPySide_QtGuiTypes = SbkPySide2_QtGuiTypes;
auto &SbkPySide_QtWidgetsTypes = SbkPySide2_QtWidgetsTypes;
auto &SbkPySide_QtPrintSupportTypes = SbkPySide2_QtPrintSupportTypes;
auto &SbkPySide_QtUiToolsTypes = SbkPySide2_QtUiToolsTypes;
```

Reply:
I see. It is just an old habit of saving as much bytes (here 8 per variable) as possible cultivated by many years of embedded development. Not sure if that counts here though...

Reply:
Perfect.

Reply:
Replaced two more `#define` with `constexpr`...